eval/src/vespa/eval/tensor/dense/dense_matmul_function.cpp

```diff
@@ -160,6 +119,35 @@ const TensorFunction &create_matmul(const TensorFunction &a, const TensorFunctio
     }
 }
 
+template <typename LCT, typename RCT, bool lhs_common_inner, bool rhs_common_inner>
+struct MyMatMulOp {
+    static auto get_fun() {
+        return my_matmul_op<LCT, RCT, lhs_common_inner, rhs_common_inner>;
+    }
+};
+
+template <bool lhs_common_inner, bool rhs_common_inner>
+struct MyMatMulOp<double, double, lhs_common_inner, rhs_common_inner> {
+    static auto get_fun() {
+            return my_cblas_double_matmul_op<lhs_common_inner, rhs_common_inner>;
+    }
+};
+
+template <bool lhs_common_inner, bool rhs_common_inner>
+struct MyMatMulOp<float, float, lhs_common_inner, rhs_common_inner> {
+    static auto get_fun() {
+        return my_cblas_float_matmul_op<lhs_common_inner, rhs_common_inner>;
+    }
+};
+
+struct MyTarget {
```

Comment:
can we eliminate this struct by letting the others capture the TypifyBool::Result instead of bool?

Reply:
alternative suggestion added below